Code/Framework/AzCore/Platform/Linux/AzCore/Debug/Trace_Linux.cpp

```diff
@@ -12,8 +12,11 @@
 
 namespace AZ::Debug::Platform
 {
-    void OutputToDebugger([[maybe_unused]] AZStd::string_view title, [[maybe_unused]] AZStd::string_view message)
+    void OutputToDebugger([[maybe_unused]] AZStd::string_view window, [[maybe_unused]] AZStd::string_view message)
     {
-        // std::cout << title << ": " << message;
+        constexpr AZStd::string_view separator = ": ";
+        fwrite(window.data(), 1, window.size(), stdout);
+        fwrite(separator.data(), 1, separator.size(), stdout);
+        fwrite(message.data(), 1, message.size(), stdout);
```

Comment:
This results in printing the logs twice on Linux.

The `Trace::RawOutput` function already performs an fwrite to the stdout FILE stream on Linux, so this code will duplicate those writes.

Reply:
While it does log to the file stream (to Server.log), it does not print to the console.  To debug linux you'd have to tail -f Server.log as well as monitor the regular (console) stdout.

Reply:
This logic should be printing to stdout on Linux https://github.com/o3de/o3de/blob/6636a3a12331365eba06d63dbe902a2968f1fb15/Code/Framework/AzCore/AzCore/Debug/Trace.cpp#L602-L613.

By default raw output goes to stdout unless the `bg_redirectrawoutput` CVAR is to a non-zero value: https://github.com/o3de/o3de/blob/6636a3a12331365eba06d63dbe902a2968f1fb15/Code/Framework/AzCore/AzCore/Debug/Trace.cpp#L93-L121

Reply:
Well, I'm not sure then, because it's not printing out to stdout.  I'm all for a better solution, but as it stands right now, most (like 99%) of the output seen in Server.log does not make it to stdout.

Reply:
Hmm, but I see what you mean, by rights I should see duplicate lines in stdout - it's effectively doing the exact same thing between OutputToDebugger and RawOutput. But between those two calls it's not getting through.
So, that's odd.

Reply:
This could be an issue with flushing the buffered stdout stream.

When messages are written to the log file, we add a newline to the message through the AZCoreLogSink OnOutput function. https://github.com/o3de/o3de/blob/6636a3a12331365eba06d63dbe902a2968f1fb15/Code/Legacy/CrySystem/AZCoreLogSink.h#L132-L175

However for RawOutput, we don't add a newline because it is "raw" output and we don't want to modify the message at all.

So what is most likely is missing is a `fflush` call to flush `stdout` FILE stream.

Reply:
You'd still expect to see the output coming from RawOutput though, right?  Because _eventually_ that fflush or newline will get added through OnOutput.  But without the change in this PR, it's not coming through at all.  

I'll dig to see what I can find

Reply:
Is the output not being duplicated to stdout?

Or actually, do you have a game.cfg or a `*.setreg` which actually sets the `bg_redirectrawoutput` to `2`, which causes raw output writes to be redirected to `/dev/null`?

Reply:
No, the output isn't duplicated at all.  I also checked both our engine repo and our project, and we don't override `bg_redirectrawoutput` at all

Reply:
I looked into it further - the logs that don't get printed to console (without this change) is as a result of 
`void CLog::LogStringToFile`

Toward the end of that function, we have explicit calls to OutputToDebugger.  Without this change, those lines never make it to stdout.

During startup there are duplicate lines printed, as Trace::Output is invoked, but that quickly stops and no repeats follow.

Reply:
Ok, in that case then the Log.cpp should update it's calls to [AZ::Debug::Platform::OutputToDebugger](https://github.com/o3de/o3de/blob/cc6f9f36df4ba97a31e07a26ed9b975c2df891a0/Code/Legacy/CrySystem/Log.cpp#L1290-L1295) with a call to `AZ::Debug::Trace::GetInstance().RawOutput()`

Reply:
RawOutput or Printf?

Additionally, there's a number of places that call OutputToDebugger outside of Log.cpp, this will end up slightly larger refactor w/ more touchpoints

Reply:
`RawOutput` should be used. The `OutputToDebugger` function doesn't format the output at all.
I am OK with restricting the changes to only `LogStringToFile`

Reply:
Using RawOutput fixes it for Linux, but then breaks Windows as it won't output to the debugger console.  Can't use "Output" either, as something swallows the output, see
`TraceMessageBus::BroadcastResult(result, &TraceMessageBus::Events::OnOutput, window, message);`

It's not immediately evident what the correct path is here.  Either Linux stdout works, or Windows.  Any suggestions?

Reply:
> `RawOutput` has the benefit of only outputting to either the stdout or stderr stream in an efficient manner and there are some case where the output might not be relevant to the debugger.

I don't know if I agree w/ this.  I checked all usages of RawOutput and I cannot see a situation where output to console would not be relevant to the debugger console.  In the case of debugging through VS, the stdout would be lost if only RawOutput (without the change addition of OutputToDebugger) was used.

Now, if there are situation where information may not be relevant, the ::Output ought to be used as it does filtering/suppression logic, which is a far more flexible/robust solution to exactly this.

Reply:
I think in the case of RawOutput being used not outputting to the Visual Studio debug window is fine, because the caller didn't ask to output the contents to the debugger.
Looking at the relevant [usages](https://github.com/search?q=repo%3Ao3de%2Fo3de+RawOutput+path%3A%2F%5ECode%5C%2F%2F&type=code) of the `Trace::RawOutput` function there are 0 cases where it needs to output to the debugger console.
There are only 2 real uses:
1. The use of `RawOutput` in [Trace_UnixLike.cpp](https://github.com/o3de/o3de/blob/dc36ec1db56e2f7c78f54568f7040cd2e2f23120/Code/Framework/AzCore/Platform/Common/UnixLike/AzCore/Debug/Trace_UnixLike.cpp#L111) is not relevant on Windows and would output using stdout on non-Windows platforms already.
2. The use in in the [Trace.cpp UnitTest](https://github.com/o3de/o3de/blob/dc36ec1db56e2f7c78f54568f7040cd2e2f23120/Code/Framework/AzCore/Tests/Debug/Trace.cpp#L164) file, There the purpose is to capture stdout and forward it to a callback WITHOUT outputting to the screen.

Reply:
> function there are 0 cases where it needs to output to the debugger console.

I think the inverse of that question is more relevant - in which cases is it explicitly _not_ needed to output to the (vs) debugger console?  As a dev, I _do_ expect to see stdout reflected in VS debugger output window already.  Note that in something like XCode, this already happens because the XCode debugger console _is_ the stdout already.  It's just for VS where we need to pipe the output explicitly (that I am aware of).

Reply:
The one case I think of when a developer wouldn't want to output to the VS debugger output window, is if they are capturing stdout to a file using `>` redirection operator such as when trying to capture all of the values of the Settings Registry via the using  `sr_regdump*` [console command](https://github.com/o3de/o3de/blob/7505ed2e96b28cc5488ff60fda0f8a0f42681921/Code/Framework/AzCore/AzCore/Settings/SettingsRegistryConsoleUtils.h#L26-L27). In those cases there is a performance hit with the actual output to the VS debug window, where the data itself isn't for debugging purposes, but informational.